observer/probers/crl/crl.go

```diff
@@ -0,0 +1,56 @@
+package probers
+
+import (
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"time"
+
+	"github.com/letsencrypt/boulder/crl/crl_x509"
+	"github.com/prometheus/client_golang/prometheus"
+)
+
+// CRLProbe is the exported 'Prober' object for monitors configured to
+// monitor CRL availability & characteristics.
+type CRLProbe struct {
+	url         string
+	cNextUpdate prometheus.GaugeVec
+	cThisUpdate prometheus.GaugeVec
+	cCertCount  prometheus.GaugeVec
+}
+
+// Name returns a string that uniquely identifies the monitor.
+func (p CRLProbe) Name() string {
+	return p.url
+}
+
+// Kind returns a name that uniquely identifies the `Kind` of `Prober`.
+func (p CRLProbe) Kind() string {
+	return "CRL"
+}
+
+// Probe requests the configured CRL and publishes metrics about it if found.
+func (p CRLProbe) Probe(timeout time.Duration) (bool, time.Duration) {
+	start := time.Now()
+	resp, err := http.Get(p.url)
+	if err != nil {
+		fmt.Println(err)
+		return false, time.Since(start)
+	}
+
+	body, err := ioutil.ReadAll(resp.Body)
+	if err != nil {
+		return false, time.Since(start)
+	}
+
+	crl, err := crl_x509.ParseRevocationList(body)
+	if err != nil {
+		return false, time.Since(start)
+	}
+
+	p.cThisUpdate.WithLabelValues(p.url).Set(float64(crl.ThisUpdate.Unix()))
+	p.cNextUpdate.WithLabelValues(p.url).Set(float64(crl.NextUpdate.Unix()))
+	p.cCertCount.WithLabelValues(p.url).Set(float64(len(crl.RevokedCertificates)))
```

Comment:
Reviewing the legacy metrics provider that this prober is replacing, I believe that the only CRL field being parsed by the legacy tool is the `next_update` timestamp. The existing tool takes the `next_update` timestamp, determines the delta between `next_update` and now and records that delta in minutes. If we want to do the same here, we can do something like:

```go
time.Since(crl.NextUpdate).Minutes()
```

This has the added benefit of returning a float64 which avoids having to do a cast here.

Reply:
The Prometheus docs consider it a best practice to export [timestamps, not times since](https://prometheus.io/docs/practices/instrumentation/#timestamps-not-time-since). This is more flexible on the 'writing alerting rules' side of things and avoids a few issues, one of which being the potential that the boulder-observer host's clock could be wrong

Reply:
Ooooh, this is neat! This is perfect fine by me, just do two things:

- Ensure that our existing tooling for establishing and checking thresholds supports this use.
- Use `crl.NextUpdate.UnixNano()` since the 32 bit timestamp returned by `.Unix()` will overflow and we've got the space for it in the float64.